sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/literals.scala

```diff
@@ -67,6 +67,7 @@ object Literal {
     case t: Timestamp => Literal(DateTimeUtils.fromJavaTimestamp(t), TimestampType)
     case d: Date => Literal(DateTimeUtils.fromJavaDate(d), DateType)
     case a: Array[Byte] => Literal(a, BinaryType)
+    case a: collection.mutable.WrappedArray[_] => apply(a.array)
```

Comment:
if you look at the doc of `Row.get`, the official type for array type is `scala.collection.Seq`

In practice, we should never match `collection.mutable.WrappedArray` explicitly. Users usually don't create this type explicitly.

Reply:
I don't think we can easily support Seq with minimised change here - let me take another look for this. That's why I targeted to support an array column in pivot specifically. Maybe I can do the type check within pivot before we call Literal.

Reply:
Yea, I'm not sure if I know the way for `Seq` here. I can make the changes inside of `pivot` to be more conservative, @cloud-fan. Let me know.

Reply:
isn't it simply `apply(seq.toArray)`?

Reply:
Nope, that's not complied..

Reply:
```
[error] /.../spark/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/literals.scala:70: No ClassTag available for _
[error]     case a: Seq[_] => apply(a.toArray)
[error]                               ^
[error] one error found
```

We can't use type tag here as well since what we need is `Any`: 

https://github.com/apache/spark/blob/86cc907448f0102ad0c185e87fcc897d0a32707f/sql/core/src/main/scala/org/apache/spark/sql/RelationalGroupedDataset.scala#L422

I know there is some subtlety. That's why I tried to explain in details ..

Reply:
Can we provide the class tag manually? like
```
apply(seq.toArray(classTag[Any]))
```

Reply:
That wouldn't work

```scala
scala> Array(1,2,3).asInstanceOf[Array[Any]]
java.lang.ClassCastException: [I cannot be cast to [Ljava.lang.Object;
  ... 29 elided

scala> classOf[Array[Any]].getComponentType()
res1: Class[_] = class java.lang.Object
```

The cases below fail to compile.


```scala
apply(a.toArray[classTag[Any]])
apply(a.toArray[scala.reflect.classTag[Any]])
```




The case below causes runtime failure.

```scala
case a: Seq[Any] => apply(a.toArray)
```

```
Unsupported component type class java.lang.Object in arrays;
org.apache.spark.sql.AnalysisException: Unsupported component type class java.lang.Object in arrays;
	at org.apache.spark.sql.catalyst.expressions.Literal$.componentTypeToDataType(literals.scala:119)
	at org.apache.spark.sql.catalyst.expressions.Literal$.apply(literals.scala:72)
	at org.apache.spark.sql.RelationalGroupedDataset.$anonfun$pivot$2(RelationalGroupedDataset.scala:427)
	at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:237)
	at scala.collection.IndexedSeqOptimized.foreach(IndexedSeqOptimized.scala:36)
	at scala.collection.IndexedSeqOptimized.foreach$(IndexedSeqOptimized.scala:33)
	at scala.collection.mutable.WrappedArray.foreach(WrappedArray.scala:39)
```